Library/Homebrew/utils/git.rb

```diff
@@ -93,8 +93,14 @@ def ensure_installed!
       if CoreTap.instance.installed?
         begin
           oh1 "Installing #{Formatter.identifier("git")}"
-          safe_system HOMEBREW_BREW_FILE, "install", "git"
-          clear_available_cache
+
+          # We need to unset `HOMEBREW_TEST_GENERIC_OS`, otherwise `git` will be
+          # installed from source in tests that need it. This is slow and will
+          # also likely fail due to `OS::Linux` and `OS::Mac` being undefined.
+          with_env "HOMEBREW_TEST_GENERIC_OS" => nil do
```

Comment:
This is a recent regression and warrants a better fix. Why was Git not being seen as available when it was?